### PR TITLE
Updated angular to v15 and other deps to latest versions. Migrated from tslint to eslint.

### PR DESCRIPTION
**Pull request by VladimirGlinskii**

### CODE REVIEW

File: common/src/auth/config.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export class AuthConfig {
   public unauthorizedRoutes: Array<string>;
-  public authService: Function;
+  public authService: any;
```
Comment (astorozhevsky) [inline common/src/auth/config.ts:3]: can we avoid any type here?

File: common/src/user/config.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export class UserConfig {
-  public userModel: { new(user: any): any };
-  public userService: Function;
+  public userModel: new(user: any) => any;
+  public userService: any;
```
Comment (astorozhevsky) [inline common/src/user/config.ts:3]: can we avoid `any` type here?

**Review by astorozhevsky — CHANGES_REQUESTED** [review]

**Review by astorozhevsky — APPROVED** [review]

**Comment by astorozhevsky** [comment]
@dmitryusenko could you please review?

File: common/src/cookie/cookie.service.ts [hunk 3]
```diff
@@ -1,165 +1,165 @@
...
+  }
+
+  protected getCookieRegExp(key: string): RegExp {
+    const escapedKey: string = key.replace(/([\[\]\{\}\(\)\|\=\;\+\?\,\.\*\^\$])/gi, '\\$1');
+
+    return new RegExp('(?:^' + escapedKey + '|;\\s*' + escapedKey + ')=(.*?)(?:;|$)', 'g');
+  }
+
+  protected getOptions(customOptions?: CookieOptions): CookieOptions {
+    return { ...this.defaultOptions, ...customOptions};
```
Comment (dmitryusenko) [inline common/src/cookie/cookie.service.ts:131]:
```suggestion
    return { ...this.defaultOptions, ...customOptions };
```

**Comment by dmitryusenko** [comment]
@VladimirGlinskii Looks good to me, thanks!

@astorozhevsky I think we can merge this.

**Pull request merged**